include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,326 @@
+
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
+        coordinate_matrix local_index_matrix{};
+
+        size_t const number_of_columns = sequence1_size + 1;
+        size_t const number_of_rows = sequence2_size + 1;
+        size_t const band_size = upper_diagonal - lower_diagonal + 1;
+
+        local_score_matrix.resize(column_index_type{number_of_columns},
+                                  row_index_type{band_size + 1},
+                                  this->lowest_viable_score());
+        local_index_matrix.resize(column_index_type{number_of_columns}, row_index_type{number_of_rows});
+
+        auto alignment_matrix_it = local_score_matrix.begin();
+        auto indexed_matrix_it = local_index_matrix.begin();
+
+        size_t row_size = std::max<int32_t>(0, -lower_diagonal);
+        size_t const column_size = std::max<int32_t>(0, upper_diagonal);
+        this->initialise_column(*alignment_matrix_it, *indexed_matrix_it, sequence2 | views::take(row_size));
+
+        // ---------------------------------------------------------------------
+        // 1st recursion phase: band intersects with the first row.
+        // ---------------------------------------------------------------------
+
+        for (auto sequence1_value : sequence1 | views::take(column_size))
+        {
+            this->compute_column(*++alignment_matrix_it,
+                                 *++indexed_matrix_it,
+                                 sequence1_value,
+                                 sequence2 | views::take(++row_size));
+        }
+
+        // ---------------------------------------------------------------------
+        // 2nd recursion phase: iterate until the end of the matrix.
+        // ---------------------------------------------------------------------
+
+        size_t first_row_index = 0u;
+        for (auto sequence1_value : sequence1 | views::drop(column_size))
+        {
+            compute_band_column(*++alignment_matrix_it,
+                                *++indexed_matrix_it | views::drop(first_row_index + 1),
+                                sequence1_value,
+                                sequence2 | views::slice(first_row_index, ++row_size));
+            ++first_row_index;
+        }
+
+        // ---------------------------------------------------------------------
+        // Final phase: track score of last column
+        // ---------------------------------------------------------------------
+
+        auto alignment_column = *alignment_matrix_it;
+        auto cell_index_column = *indexed_matrix_it | views::drop(first_row_index);
+
+        auto alignment_column_it = alignment_column.begin();
+        auto cell_index_column_it = cell_index_column.begin();
+
+        this->track_last_column_cell(*alignment_column_it, *cell_index_column_it);
+
+        for ([[maybe_unused]] auto && unused : sequence2 | views::slice(first_row_index, row_size))
```

Comment:
this has no performance regression compared to a simple `for (size_t i = first_row_index; i < first_row_index + row_size; ++i)`?

Reply:
It is not about performance. It is about beautiful code ;)

Reply:
beauty lies in the eye of the beholder :)